### ISSUE TITLE
[Time field] Functionality needs to be clarified to assistive technologies (and users)

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2866]
### Provide a general summary of the issue here

The time field is a bit confusing to use with both keyboard and screen reader. The field acts somewhat similar to `<input type="time">` but does not provide clear information on the elements functions (e.g., missing roles or instructions), and due to its implementation, some assistive technologies might not work great with it.

### 🤔 Expected Behavior?

The field works similar to `<input type="time">` or the functions are otherwise clear and operable by assistive technology and assistive technology users. 

### 😯 Current Behavior

The behaviour seems to be created using HTMLInputElement selectionStart, but since the behaviour will always only include either the hours or minutes, and never the whole text input, and editing only single characters requires some fancy keyboard work (doable, but annoying), it makes this confusing for users and also assistive technology itself. (Yes, users can select the whole text with shortcuts, but the idea is about clarity to users).

This implementation might cause some wonky behaviour in some cases:
![Image](https://github.com/user-attachments/assets/a9e20f9e-a3c5-40e7-b49f-cab6dddef690)

**With NVDA:** When a screen reader user first enters the field, the field is an editable textfield:
<img width="256" height="112" alt="Text field labelled Time" src="https://github.com/user-attachments/assets/77947bbe-8b0b-4d6d-aa80-ea01c47c1116" />

After entering the time, the editable textfield changes (functionally) to two spinbuttons with no role or a programmatic name:
<img width="266" height="119" alt="Text field labelled Time showing 21:30. The time " src="https://github.com/user-attachments/assets/84f115cb-0d1d-4a40-87dd-39c574aeef5f" />

Spinbutton is an element in which users can increase or decrease a value using up or down arrow keys, or edit (if implemented) the value. This is basically the `<input type="number">`, but elements with role="spinbutton" can take many forms.

This could be confusing, because the user would expect to return to a textfield with full text selected, rather than only part of the text being editable. The spinbuttons don't have a role or an accessible name either, nor is the minutes tabbable, which might make some users wonder, where the last part of the time went, even though the textfield communicates the time correctly. The text selection part is most probably not a violation, as text selection is a part of a textfield, but the parts of the text having actions, which is not native to text field, definitely needs some clarification in one way or another. 

The textfield also changes to spinbutton (functionally) when user presses arrow down or up keys.

**For Android users** (especially TalkBack users): Since users might not have the benefit of the arrow keys, they can easily miss that the hours and minutes are separate: 
<img width="681" height="378" alt="Text field has focus on hours text only when using talkback" src="https://github.com/user-attachments/assets/63151ddf-5a2c-4d56-b7c5-88c0bb2256a1" />

Which could lead to this situation (though, the field will clear on invalid input, which probably just confuses users more at this point):
<img width="627" height="660" alt="Time field reads 04:45:30 because user entered the whole time incorrectly to the slot that only fits hours" src="https://github.com/user-attachments/assets/8154130d-f469-434f-b26c-2acfc7d65a00" />

The latter part of the text is editable, and after removing the text selection (i.e., swiping up or down to move the caret), the text field functions like a typical text field.

**iOS users using VoiceOver** cannot edit the text field at all. Selecting the text field opens the dropdown list, and VoiceOver announces incorrectly "collapsed", though the text field itself does not communicate in any way that it would be a combobox. Users can navigate the list to select time, but cannot use the text field itself. To my best knowledge, VoiceOver emulates tapping with finger (rather than, e.g., keyboard focus actions), which is why the combobox opens. VoiceOver also tends to group elements together, which is why some of the functionality isn't focusable, as it might treat them as nested behaviours. I assume VO might actually group everything under the <sl-timefield> tag, and therefore all the elements (text field, menu button, and the listbox) are grouped together, and therefore only some of the functionality is actually focusable using VO.

### 💁 Possible Solution

- The behaviour is created using selectionStart, but for assistive technology users (screen reader users) this should rather be implemented as two different spinbuttons (or comboboxes with separate dropdown lists) 
- both spinbuttons should exist before user enters the field (similar to a `<input type="time">`):
<img width="553" height="160" alt="Image" src="https://github.com/user-attachments/assets/9bd30868-072e-47a3-a2c8-06ff6710a463" />.
- The spinbuttons should have role="spinbutton"
- The spinbuttons should have accessible names (e.g., "hours" and "minutes")
- The spinbuttons should be tabbable (tabindex="0"), so users cannot accidentally miss the minutes field (currently selectable only with arrow keys)

[Spinbutton example using WAI-ARIA (W3C ARIA Authoring practices guide)](https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/examples/quantity-spinbutton/)

OR

The textfield could just be a textfield, with the menu/listbox as an additional element (with instruction on the time format in text, so users know what to input).

OR

You could try to explain the functions to users in text. (Yes, this is a valid option, but an extremely desperate one)

### 🖥️ Steps to Reproduce

NVDA:
1. Set focus to the text field -> reads "Placeholder, edit, blank"
2. Add time by either selecting from the dropdown list, or text input (e.g., 21:30)
3. Move focus to another element, then back to the text field -> reads "Placeholder, edit, selected 21"
4. Use arrow right -> reads "blank, 30 selected, 21 unselected"
4. Use arrow left-> reads "colon, 21 selected, 30 unselected"
5. Use arrow up -> reads "22:15"

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome, Safari, Firefox

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by anna-lach** [comment]
Some comments after meeting with Avaava:

- We should use separate inputs with `spinbutton` role for hours and minutes,
- Those separate inputs should be focusable,
- When someone has typed in `hour`, the focus should automatically jump from hours to minutes,
- Time field should swallow the alphabet (only numbers should be available to type in),
- When the time-field is invalid, the validation should be shown later, not while typing.

Please also take a look at the example:
https://mui.com/x/react-date-pickers/time-field/

`Time field` and `date field` should work in a similar way and should be built in a similar way too.
